### DISCUSSION TITLE
기존 EKS 설정 파일 수정 및 오류 해결

### DISCUSSION THREAD

**Discussion by ssedogdoom** [cloud2scape/.github#40] [cloud-일반]
### 내용

![image](https://github.com/user-attachments/assets/7baebcc7-7223-4ef2-b1d4-edfa02e74325)

EKS 설정 파일(`eks.tf`)과 VPC 설정 파일(`vpc.tf`)의 오류를 해결하기 위해 아래와 같이 수정 작업을 진행했습니다:

- `eks.tf` 파일:
  - 잘못된 변수 호출 선언 문 수정
  - IAM 관련 생성 섹션 삭제 (모듈에서 자동 생성됨)
  - Security Group(SG) 관련 섹션은 임시로 삭제
  - 서브넷 및 VPC ID 호출 방식을 변경하여 정확한 값 호출
  - `local` 변수를 활용해 코드 구성 간소화

- `vpc.tf` 파일:
  - 특정 ID 값을 불러오는 방식이 `module.~~~` 형식으로 되어 있었는데, 이를 수정하여 오류를 해결

### 문제

위와 같은 수정으로 로컬 환경에서는 문제가 없었으나, 다른 사람의 컴퓨터에서는 여전히 오류가 발생했습니다. 
![image](https://github.com/user-attachments/assets/38ba0446-ec8a-4de4-b8a2-6be131269bf6)

주요 원인으로는 다음과 같은 요소가 의심됩니다:

1. 삭제가 제대로 이루어지지 않아 이전 설정을 참조하는 문제
2. KMS 설정에 따른 충돌 (새로 배포할 때 KMS 이름을 다르게 하면 문제 해결 가능)

제 생각에는 첫 번째 원인(이전 설정 참조)이 주요 원인으로 판단됩니다.

### 해결 방안 제안

오류 해결을 위해 아래의 절차를 제안합니다:

1. 기존 설정을 완전히 **DESTROY**
2. **REPOSITORY CLONE**으로 새롭게 설정 복사
3. **AWS CLI**를 통해 로그인
4. `vpc.tf`, `eks.tf` 파일이 있는 디렉토리에서 작업
5. `terraform init` 및 `terraform plan` 실행하여 오류 확인
6. 오류가 없다면 `terraform apply`로 적용

### 추가 조치 

혹시 모를 문제를 대비해, 다른 디렉토리에 새로운 코드를 작성하여 이를 BASE로 기존 코드와 결합해 커스터마이징과 남은 인프라를 구축 진행할 예정입니다. 

### 결론
현재 새로운 코드로 EKS가 어느 환경에서나 정상적으로 작동하는 것을 확인했으며, 이후 애플리케이션 포팅 작업이 완료된다면 웹을 통한 애플리케이션 접속 및 배포 작업을 진행할 수 있을 것으로 예상됩니다.

즉, 현재는 EKS만 정상적으로 올라가 있는 상태이며, 포팅 작업을 통해 최종적으로 웹 접속이 가능해지도록 설정할 계획입니다.

**Comment by laigasus** [comment]
현재 제공하신 코드는 단일 워커노드에서만 동작하며,
external ip 가 생성되지 않습니다.
참고하세용

```
~\IdeaProjects\Infra git:[dev]
kubectl get all
NAME                                   READY   STATUS    RESTARTS   AGE
pod/core-gateway-57f86b9f4-xsdk2       1/1     Running   0          2m21s
pod/service-product-56fc99f644-6qwq9   1/1     Running   0          2m25s

NAME                      TYPE           CLUSTER-IP       EXTERNAL-IP   PORT(S)        AGE
service/core-gateway      LoadBalancer   172.20.102.218   <pending>     80:30908/TCP   2m21s
service/kubernetes        ClusterIP      172.20.0.1       <none>        443/TCP        56m
service/service-product   NodePort       172.20.53.80     <none>        80:32003/TCP   2m25s

NAME                              READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/core-gateway      1/1     1            1           2m21s
deployment.apps/service-product   1/1     1            1           2m25s

NAME                                         DESIRED   CURRENT   READY   AGE
replicaset.apps/core-gateway-57f86b9f4       1         1         1       2m21s
replicaset.apps/service-product-56fc99f644   1         1         1       2m25s
```

**Comment by ssedogdoom** [comment]
해결 다 했고 구축 성공했습니다.
